libgnucash/backend/sql/gnc-recurrence-sql.cpp

```diff
@@ -164,8 +164,9 @@ get_recurrence_period_type (gpointer pObject)
     g_return_val_if_fail (pObject != NULL, NULL);
     g_return_val_if_fail (pInfo->pRecurrence != NULL, NULL);
 
-    return (gpointer)recurrencePeriodTypeToString (
-               recurrenceGetPeriodType (pInfo->pRecurrence));
+    return (gpointer)g_strdup (recurrencePeriodTypeToString
+                               (recurrenceGetPeriodType
+                                (pInfo->pRecurrence)));
```

Comment:
I mean this change preserves the `strdup`ed string. Is it truly necessary? I'd test by saving a book with SX into SQLite3 but I can't seem to enable sqlite3.

Reply:
I don't think it is. It's eventually stuffed into a std::string without a subsequent free so making it const char* should be right.

Reply:
> I don't think it is. It's eventually stuffed into a std::string without a subsequent free

Ok where's that?

Reply:
[here](https://github.com/Gnucash/gnucash/blob/4b6a9b85605e3cd9adecf2c60a839fd97c45226a/libgnucash/backend/sql/gnc-sql-column-table-entry.cpp#L153) via [here](https://github.com/Gnucash/gnucash/blob/4b6a9b85605e3cd9adecf2c60a839fd97c45226a/libgnucash/backend/sql/gnc-sql-column-table-entry.hpp#L349), `get_recurrence_period_type` etc. having been passed as the `getter`.